dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -464,203 +542,216 @@ protected void loadPermissions(final Contentlet con, final Map<String,Object> m)
-								+ ESMappingConstants.PERSONAS, personaTagsNames);
-						m.put(ESMappingConstants.PERSONAS, personaTagsNames);
+						continue;
 					}
-
-				} else if(f.getFieldType().equals(CUSTOM_FIELD.legacyValue())
-						&& f.getVelocityVarName().equals(PERSONA_KEY_TAG_FIELD_VAR)) {
-					m.put(PERSONA_KEY_TAG,valueObj.toString());
-					m.put(keyName, valueObj.toString());
+				} else if(field.getFieldType().equals(CUSTOM_FIELD.legacyValue())
+						&& field.getVelocityVarName().equals(PERSONA_KEY_TAG_FIELD_VAR)) {
+					contentletMap.put(PERSONA_KEY_TAG,valueObj.toString());
+					contentletMap.put(keyName, valueObj.toString());
 				} else {
-					if (f.getFieldContentlet()
+					if (field.getFieldContentlet()
 							.startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_BOOLEAN)) {
-						m.put(keyName, valueObj);
-						m.put(keyNameText,valueObj.toString());
-					} else if (f.getFieldContentlet()
-							.startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_FLOAT) || f
+						contentletMap.put(keyName, valueObj);
+						contentletMap.put(keyNameText,valueObj.toString());
+					} else if (field.getFieldContentlet()
+							.startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_FLOAT) || field
 							.getFieldContentlet()
 							.startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_INTEGER)) {
-						m.put(keyName, valueObj);
-						m.put(keyNameText, numFormatter.format(valueObj));
+						contentletMap.put(keyName, valueObj);
+						contentletMap.put(keyNameText, numFormatter.format(valueObj));
 					} else {
-						m.put(keyName, valueObj);
-						m.put(keyNameText, valueObj.toString());
+						contentletMap.put(keyName, valueObj);
+						contentletMap.put(keyNameText, valueObj.toString());
 					}
 				}
 
 				// Store sha256 hash for unique fields in the index
-				if (f.isUnique() && m.containsKey(keyName)) {
-					final Object uniqueValue = m.get(keyName);
-					m.put(keyName + ESUtils.SHA_256,
-							ESUtils.sha256(keyName, uniqueValue, con.getLanguageId()));
+				if (field.isUnique() && contentletMap.containsKey(keyName)) {
+					final Object uniqueValue = contentletMap.get(keyName);
+					contentletMap.put(keyName + ESUtils.SHA_256,
+							ESUtils.sha256(keyName, uniqueValue, contentlet.getLanguageId()));
 				}
-
 			} catch (Exception e) {
-				Logger.warn(ESMappingAPIImpl.class, "Error indexing field: " + f.getFieldName()
-						+ " of contentlet: " + con.getInode(), e);
+				Logger.warn(ESMappingAPIImpl.class, "Error indexing field: " + field.getFieldName()
+						+ " of contentlet: " + contentlet.getInode(), e);
 				throw new DotDataException(e.getMessage(),e);
 			}
 		}
 	}
 
+	private boolean loadTagsField(final Contentlet contentlet,
+								  final Map<String, Object> contentletMap,
+								  final Structure structure,
+								  final String keyName) throws DotDataException {
+
+		final List<Tag> tagList = APILocator.getTagAPI().getTagsByInode(contentlet.getInode());
+		if(tagList ==null || tagList.size()==0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)